quote-generator/src/app/app.functions/create-quote.js

```diff
@@ -0,0 +1,98 @@
+// for HubSpot API calls
+const hubspot = require('@hubspot/api-client');
+
+const hubspotClient = new hubspot.Client({
+  accessToken: process.env['PRIVATE_APP_ACCESS_TOKEN'],
+});
+
+exports.main = async (context = {}, sendResponse) => {
+  const { hs_object_id } = context.propertiesToSend;
+  const { distance, sku, numberOfBuses, quoteName } = context.event.payload;
+
+  const product = await findProductBySKU(sku);
+
+  if (product === null) {
+    sendResponse({ error: 'PRODUCT_NOT_FOUND' });
+  } else {
+    const quote = await createQuote({
+      dealId: hs_object_id,
+      quoteName,
+    });
+    for (let i = 0; i < numberOfBuses; i++) {
+      await addLineItem({
```

Comment:
Do you need to await that each promise resolves before calling the next one? If the answer is not, I would consider creating an array of promises inside the for loop and then using `Promise.all` to resolve them in parallel